app/src/ui/lib/toggletipped-content.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react'
+import { ITooltipProps, Tooltip } from './tooltip'
+import { createObservableRef } from './observable-ref'
+import classNames from 'classnames'
+import { AriaLiveContainer } from '../accessibility/aria-live-container'
+
+/**
+ * IToggledtippedContentProps is a superset of ITooltipProps but does not
+ * define the `target` prop as that's set programatically in render
+ */
+interface IToggledtippedContentProps
+  extends Omit<ITooltipProps<HTMLElement>, 'target'> {
+  /** The tooltip contents */
+  readonly tooltip: JSX.Element | string | undefined
+
+  /**
+   * An optional additional class name to set on the tooltip in order to be able
+   * to apply specific styles to the tooltip
+   */
+  readonly tooltipClassName?: string
+
+  /** An optional class name to set on the wrapper element */
+  readonly className?: string
+
+  /** An optional aria-label property in case children is not descriptive
+   * - for example an icon */
+  readonly ariaLabel?: string
+}
+
+interface IToggledtippedContentState {
+  /** State of when the tooltip is visible */
+  readonly tooltipVisible: boolean
+}
+
+/**
+ * A less keyboard toggleable version of the Tooltip content component for when
+ * it's acceptable to add a wrapping element around the content. The content
+ * will be wrapped in a button as it is toggleable. supports all the options
+ * that the Tooltip component does without having to worry about refs.
+ **/
+export class ToggledtippedContent extends React.Component<
+  IToggledtippedContentProps,
+  IToggledtippedContentState
+> {
+  private buttonRef = createObservableRef<HTMLButtonElement>()
+  private shouldForceAriaLiveMessage = false
+
+  public constructor(props: IToggledtippedContentProps) {
+    super(props)
+    this.state = {
+      tooltipVisible: false,
+    }
+
+    this.buttonRef.subscribe(this.onButtonRef)
+  }
+
+  private onToggle = () => {
+    this.shouldForceAriaLiveMessage = !this.shouldForceAriaLiveMessage
+  }
+
+  private onTooltipVisible = () => {
+    this.setState({ tooltipVisible: true })
+  }
+
+  private onTooltipHidden = () => {
+    this.setState({ tooltipVisible: false })
+  }
+
+  private onButtonRef = (ref: HTMLButtonElement | null) => {
+    if (ref === null) {
+      const currRef = this.buttonRef.current
```

Comment:
Unfortunately I don't think this will work or at least I would be very surprised if it did. The `current` property should be updated before the subscribers are called IIRC (or at least that would make the most sense).

So I think you're gonna have to keep a copy of the ref as an instance variable as well.

Reply:
۱